### PR TITLE
Add User contributed docset

### PR DESCRIPTION
**Pull request by troopa81**
Add the capacity to see and download Dash user contributed Docset (ticket #170).

I try my best to keep the way docset download was managed (cache for instance) and to follow the coding rules (editorconfig). 

Regards,
Julien

### CODE REVIEW

File: src/libs/ui/docsetsdialog.cpp [hunk 1]
```diff
@@ -59,6 +59,13 @@ const char ApiServerUrl[] = "http://api.zealdocs.org/v1";
 const char RedirectServerUrl[] = "https://go.zealdocs.org";
 // TODO: Each source plugin should have its own cache
 const char DocsetListCacheFileName[] = "com.kapeli.json";
+const char UserContributedDocsetListCacheFileName[] = "com.kapeli.json.user_contributed";
+
+// TODO manage mirror
+const char UserContributedApiUrl[] =
+  "http://london.kapeli.com/feeds/zzz/user_contributed/build/index.json";
+const char UserContributedDocsetListUrl[] =
+  "http://london.kapeli.com/feeds/zzz/user_contributed/build/%1/%2";
```
Comment (sighingnow) [inline src/libs/ui/docsetsdialog.cpp]: Maybe we should use `http://kapeli.com/feeds/zzz/user_contributed/build/index.json` and `http://kapeli.com/feeds/zzz/user_contributed/build/%1/%2`. for better CDN usage.

Comment (Kapeli) [reply src/libs/ui/docsetsdialog.cpp]: "kapeli.com" just goes to the server that hosts my website, which is in the same datacenter as newyork.kapeli.com. I prefer if you don't use kapeli.com but one of the other mirrors.

Comment (sighingnow) [reply src/libs/ui/docsetsdialog.cpp]: I see, I had expected "kapeli.com" has geo-based DNS solver, pointing to mirrors in other regions. Thanks for the clarification.

**Comment by troopa81** [comment]
Just a gently reminder for this PR to be merged

**Comment by trollixx** [comment]
Thanks for the contribution, and sorry for the delayed response, lately I haven't had much time for Zeal.

I don't really want to merge this, because my idea is that Zeal should only talk to `api.zealdocs.org`. That gives much more control, and should allow unified approach for handling all docsets on the client side.

I hoped to make changes on the server-side rather soon, but if that will not happen in time for the next release, I'll merge this PR as a temporary solution.

**Comment by troopa81** [comment]
I understand your will to manage this on server side.

Zeal is a great tool and the only thing preventing me from using it more widely was the access to user docset. 

I keep an eye on the project waiting for server side modifications so. If you need to merge this PR for temporary solution and need some modifications, just let me know.

**Comment by ryanpcmcquen** [comment]
@trollixx, any progress here?

**Pull request closed without merging**